modules/node/src/channel/channel.service.ts

```diff
@@ -188,7 +194,13 @@ export class ChannelService {
     this.log.info(
       `Rebalance finished for ${channel.userIdentifier} on chain ${channel.chainId}, assetId: ${assetId}`,
     );
-    return response;
+    return (
+      depositRes && {
+        transaction: depositRes.tx,
+        depositAppIdentityHash: depositRes.appIdentityHash,
+        rebalanceType,
+      }
+    );
```

Comment:
will this properly return `undefined` or will it return `false`?

Reply:
If `depositRes` is undefined when it's set above (which it should be), it will hit the first half of the statement, and since undefined is falsy, it'll short circuit and return undefined there.